pkg/cybr/conjur/authenticators/iam/authenticate.go

```diff
@@ -0,0 +1,66 @@
+package iam
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+func getAuthnURL(authnURL string, account string, login string) string {
+	identifier := url.QueryEscape(login)
+	return fmt.Sprintf("%s/%s/%s/authenticate", authnURL, account, identifier)
+}
+
+func newHTTPSClient(ignoreSSLVerify bool, cert []byte) (*http.Client, error) {
+	// If not certificate provided do not create a certifictae pool
+	if cert == nil {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: ignoreSSLVerify},
+		}
+		return &http.Client{Transport: tr, Timeout: time.Second * 10}, nil
+	}
```

Comment:
If no client side certificate is provided then default to InsecureSkipVerify? IDK if this should be the expected behavior. Currently Ignoring verify would leave the client susceptible to a MITM.

If you want this behavior it should probably be another argument like `ignoreSSLVerify`.

Reply:
This is fixed in the latest commit.